geopandas/geodataframe.py

```diff
@@ -1608,12 +1655,10 @@ def explode(self, column=None, **kwargs):
 
         exploded_geom = df_copy.geometry.explode().reset_index(level=-1)
         exploded_index = exploded_geom.columns[0]
-
-        df = (
-            df_copy.drop(df_copy._geometry_column_name, axis=1)
-            .join(exploded_geom)
-            .__finalize__(self)
-        )
+        df = GeoDataFrame(
+            df_copy.drop(df_copy._geometry_column_name, axis=1).join(exploded_geom),
+            geometry=self._geometry_column_name,
+        ).__finalize__(self)
```

Comment:
> The reason for this change is because drop now returns a DataFrame instead of GeoDataFrame?

That's exactly right. I was thinking that using the constructor leaves less ambiguity about the type of drop - but I guess if this PR goes in then that will be standard and there is no ambiguity.

Is there potential implications for extra copies with one versus the other?  - perhaps not relevant here, but as a pattern in general? I don't know enough about pandas internals, but I have seenthe monkey patched `pd.DataFrame.set_geometry` forces a copy.

Reply:
Hmm, yeah you're right. `set_geometry` will result in an extra copy (with current pandas, I have a proposal to change that though ;)) 
I personally like the code using `set_geometry` more, but so that might be a reason to keep it as you had originally with GeoDataFrame(..)